src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -12,14 +12,21 @@ namespace System.Text.RegularExpressions
     internal sealed class RegexInterpreterFactory : RegexRunnerFactory
     {
         private readonly RegexInterpreterCode _code;
+        private readonly CultureInfo? _culture;
 
-        public RegexInterpreterFactory(RegexTree tree, CultureInfo culture) =>
+        public RegexInterpreterFactory(RegexTree tree, CultureInfo culture)
+        {
+            // We use the culture field from the tree instead of the passed in culture because the field on the tree
+            // will only be set to an actual culture if the tree contains IgnoreCase backreferences. If the tree doesn't
+            // have IgnoreCase backreferences, then we keep _culture as null.
+            _culture = tree.Culture;
             // Generate and store the RegexInterpretedCode for the RegexTree and the specified culture
             _code = RegexWriter.Write(tree, culture);
```

Comment:
Why does RegexWriter.Write need the culture?  And if it needs the culture, why can't it use tree.Culture?  Either way we could get rid of the culture parameter to RegexInterpreterFactory.